### DISCUSSION TITLE
Nested Variables in include, metadata

### DISCUSSION THREAD

**Discussion by tmeuze** [fralau/mkdocs-macros-plugin#93] [Q&A] [answered]
In order to avoid a ton of HTML in my markdown files, I am trying to sort of nest variables. Please bear with me as I have almost no clue what I'm doing. 

I have HTML code/files that are all unique to every `nav_item`/page. Right now, I have all of the HTML codes placed in their own metadata tag, for each page, and it works - but it's ugly. What I would like to do is `include` them programmatically. 

I've tried two approaches. In HTML, I've tried something along the lines of:
```
{% include "path/to/{{ nav_item.meta.uniquename }}" %}
```
Where `uniquename` is a custom meta tag specifying the HTML file.

The other approach was to have this in HTML:
```
{{nav_item.meta.uniquename}}
```
Then a metatag on each page specifies what HTML file to include:
```
uniquename:
    {% include "path/to/uniquename.html" %}
```

But this doesn't get parsed. It outputs verbatim `{% include "path/to/uniquename.html" %}`

So it appears that "nested" variables aren't supported, that an `include` directive can't be placed in a metatag, or that I'm completely doing it wrong. 

If anyone has any guidance, I would greatly appreciate it. I hope this post was clear.

**Comment by fralau** [comment, marked as answer]
That does not seem too difficult. 

Could you try this?

`{% include "path/to/" +  nav_item.meta.uniquename %}`

There should be no need to "nest", since the content of curly braces is more or less Python syntax.

(I am not sure what `nav_item` is, but I trust it is a good page descriptor?)

**Reply by tmeuze** [reply]
Wow, thanks so much for the quick response! I have tried this; let me tell include the specifics. 

My HTML files are in the `partials/bignav` directory. So in HTML I have:
`{% include "partials/bignav/" + nav_item.meta.bignav %}`

As a metatag for each root page, I have:
`bignav: msp.html`

Where `msp.html` is the file I want to include. 

What seems to be happening is it takes that literally; the output ends up being the contents of `msp.html`. I can see in the serve logs, it's trying to find the file: `partials/bignav/<HTML CODE...>`

Thanks, again.

PS - I believe `nav_item.meta` is similar to `page.meta`, but is rendered for the navigation item, rather than on load of a page.

**Reply by tmeuze** [reply]
Actually, the above may have been a human error. I am now getting this instead:
`jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'bignav'`

**Reply by fralau** [reply]
It means that `nav_item.meta` des not have any attribute `bignav`?

At this point you might need to be more specific. Could you give a sample of the markdown page, etc?

**Reply by tmeuze** [reply]
Of course. You're right, I was testing that attribute and didn't know it needed to be applied to all top-level pages. Everything is working as expected now. 

Thanks again for your help and time.
